### PR TITLE
*: fix 'gosimple', 'gounused' checks

### PR DESCRIPTION
**Pull request by gyuho**
To clean up Travis outputs...

### CODE REVIEW

**Comment by heyitsanthony** [comment]
`test` probably needs some changes to error out if these weren't causing CI to fail

**Comment by gyuho** [comment]
@heyitsanthony @xiang90 Go build cmd seems like just returning 0 for build fails

https://github.com/golang/go/blob/master/src/go/build/build.go#L675

```
func exit() {
	for _, f := range atexitFuncs {
		f()
	}
	os.Exit(exitStatus) // exitStatus is 0
}
```

I tried to grep stderr

```
	if which gosimple >/dev/null; then
		echo "Checking gosimple..."
		for path in $GOSIMPLE_UNUSED_PATHS; do
			simplResult=`gosimple $REPO_PATH/${path} 2>&1`
			if [ -n "${simplResult}" ]; then
				echo -e "gosimple checking ${path} failed:\n${simplResult}"
				exit 255
			fi
		done
	else
		echo "Skipping gosimple..."
	fi
```

but this doesn't print out the actual error logs so makes it hard to debug; see here https://travis-ci.org/coreos/etcd/jobs/182730600.

All other errors are fixed. https://travis-ci.org/coreos/etcd/jobs/182731343#L508

Thanks.

**Comment by xiang90** [comment]
lgtm

**Comment by gyuho** [comment]
@heyitsanthony Would this break anything in grpcproxy? https://jenkins-etcd.prod.coreos.systems/job/etcd-proxy/262/ doesn't tell much about failures. Can you retrigger?

**Comment by gyuho** [comment]
@heyitsanthony Ok all fixed.

I made it fail with build and test fails as it should https://travis-ci.org/coreos/etcd/jobs/182752373
I made it fail with the new script and works as expected  https://travis-ci.org/coreos/etcd/jobs/182750950

File: test [hunk 1]
```diff
@@ -21,7 +21,7 @@ fi
 
 # TODO: 'client' pkg fails with gosimple from generated files
 # TODO: 'rafttest' is failing with unused
-GOSIMPLE_UNUSED_PATHS=$(go list ./... | sed -e 's/github.com\/coreos\/etcd\///g' | grep -vE 'cmd|vendor|rafttest|github.com/coreos/etcd$|client$')
+GOSIMPLE_UNUSED_PATHS=$(go list ./... | grep 'github.com/coreos/etcd' | sed -e 's/_\/home\/travis\/gopath\/src\///g' | sed -e 's/github.com\/coreos\/etcd\///g' | grep -vE 'cmd|vendor|rafttest|github.com/coreos/etcd$|client$')
```
Comment (heyitsanthony) [inline test]: `grep 'github.com/coreos/etcd' ` is causing failures on the proxy CI. Is it necessary?

Comment (gyuho) [inline test]: Not really. Just removed. Let's see if it works now. Thx.

**Comment by gyuho** [comment]
@heyitsanthony It fails if I change the script. https://travis-ci.org/coreos/etcd/jobs/182764154

Do we know why it fails in jenkins?

**Comment by heyitsanthony** [comment]
@gyuho The jenkins path does an out of gopath build so `github.com/coreos/etcd` doesn't show up in `go list ./...`. I don't understand why the travis job is fails without the grep-- what is the grep stripping away for travis? Anyway, the test script probably shouldn't be travis-aware, would using `pwd` be enough so it works for all cases?

If it can't work in both environments, I can set jenkins to ignore the fmt pass, but GOSIMPLE_UNUSED_PATHS will need to move into the fmt_pass function (which is probably a better place for it, regardless).

CI seems to be posting the regress-pr results and not the proxy CI results so it looks like it passes when it doesn't. Ugh.

**Comment by gyuho** [comment]
So the problem with travis is

If you look at this https://travis-ci.org/coreos/etcd/jobs/182700096 where I ran `go list ./...`

```
go list ./...
_/home/travis/gopath/src/github.com/coreos/etcd
_/home/travis/gopath/src/github.com/coreos/etcd/alarm
_/home/travis/gopath/src/github.com/coreos/etcd/auth
_/home/travis/gopath/src/github.com/coreos/etcd/auth/authpb
_/home/travis/gopath/src/github.com/coreos/etcd/client
_/home/travis/gopath/src/github.com/coreos/etcd/client/integration
_/home/travis/gopath/src/github.com/coreos/etcd/clientv3
_/home/travis/gopath/src/github.com/coreos/etcd/clientv3/concurrency
_/home/travis/gopath/src/github.com/coreos/etcd/clientv3/integration
_/home/travis/gopath/src/github.com/coreos/etcd/clientv3/mirror
_/home/travis/gopath/src/github.com/coreos/etcd/clientv3/naming
github.com/beorn7/perks/quantile
github.com/bgentry/speakeasy
github.com/boltdb/bolt
github.com/cockroachdb/cmux
github.com/coreos/go-semver/semver
github.com/coreos/go-systemd/daemon
github.com/coreos/go-systemd/journal
github.com/coreos/go-systemd/util
github.com/coreos/pkg/capnslog
github.com/coreos/pkg/dlopen
github.com/cpuguy83/go-md2man/md2man
github.com/davecgh/go-spew/spew
github.com/dustin/go-humanize
github.com/ghodss/yaml
github.com/gogo/protobuf/proto
github.com/golang/groupcache/lru
github.com/golang/protobuf/jsonpb
github.com/golang/protobuf/proto
github.com/google/btree
github.com/grpc-ecosystem/go-grpc-prometheus
github.com/grpc-ecosystem/grpc-gateway/runtime
github.com/grpc-ecosystem/grpc-gateway/runtime/internal
github.com/grpc-ecosystem/grpc-gateway/utilities
github.com/inconshreveable/mousetrap
github.com/jonboulle/clockwork
github.com/kr/pty
github.com/mattn/go-runewidth
github.com/matttproud/golang_protobuf_extensions/pbutil
github.com/olekukonko/tablewriter
github.com/pmezard/go-difflib/difflib
github.com/prometheus/client_golang/prometheus
github.com/prometheus/client_model/go
github.com/prometheus/common/expfmt
github.com/prometheus/common/internal/bitbucket.org/ww/goautoneg
github.com/prometheus/common/model
github.com/prometheus/procfs
github.com/russross/blackfriday
github.com/shurcooL/sanitized_anchor_name
github.com/spf13/cobra
github.com/spf13/pflag
github.com/stretchr/testify/assert
github.com/ugorji/go/codec
github.com/urfave/cli
github.com/xiang90/probing
golang.org/x/crypto/bcrypt
golang.org/x/crypto/blowfish
golang.org/x/net/context
golang.org/x/net/http2
golang.org/x/net/http2/hpack
golang.org/x/net/internal/timeseries
golang.org/x/net/trace
golang.org/x/sys/unix
golang.org/x/time/rate
google.golang.org/grpc
google.golang.org/grpc/codes
google.golang.org/grpc/credentials
google.golang.org/grpc/grpclog
google.golang.org/grpc/internal
google.golang.org/grpc/metadata
google.golang.org/grpc/naming
google.golang.org/grpc/peer
google.golang.org/grpc/transport
gopkg.in/cheggaaa/pb.v1
gopkg.in/yaml.v2
_/home/travis/gopath/src/github.com/coreos/etcd/compactor
_/home/travis/gopath/src/github.com/coreos/etcd/contrib/raftexample
_/home/travis/gopath/src/github.com/coreos/etcd/contrib/recipes
_/home/travis/gopath/src/github.com/coreos/etcd/contrib/systemd/etcd2-backup-coreos
_/home/travis/gopath/src/github.com/coreos/etcd/discovery
_/home/travis/gopath/src/github.com/coreos/etcd/e2e
_/home/travis/gopath/src/github.com/coreos/etcd/embed
_/home/travis/gopath/src/github.com/coreos/etcd/error
_/home/travis/gopath/src/github.com/coreos/etcd/etcdctl
_/home/travis/gopath/src/github.com/coreos/etcd/etcdctl/ctlv2
_/home/travis/gopath/src/github.com/coreos/etcd/etcdctl/ctlv2/command
_/home/travis/gopath/src/github.com/coreos/etcd/etcdctl/ctlv3
_/home/travis/gopath/src/github.com/coreos/etcd/etcdctl/ctlv3/command
_/home/travis/gopath/src/github.com/coreos/etcd/etcdmain
_/home/travis/gopath/src/github.com/coreos/etcd/etcdserver
_/home/travis/gopath/src/github.com/coreos/etcd/etcdserver/api
_/home/travis/gopath/src/github.com/coreos/etcd/etcdserver/api/v2http
_/home/travis/gopath/src/github.com/coreos/etcd/etcdserver/api/v2http/httptypes
_/home/travis/gopath/src/github.com/coreos/etcd/etcdserver/api/v3rpc
_/home/travis/gopath/src/github.com/coreos/etcd/etcdserver/api/v3rpc/rpctypes
_/home/travis/gopath/src/github.com/coreos/etcd/etcdserver/auth
_/home/travis/gopath/src/github.com/coreos/etcd/etcdserver/etcdserverpb
_/home/travis/gopath/src/github.com/coreos/etcd/etcdserver/membership
_/home/travis/gopath/src/github.com/coreos/etcd/etcdserver/stats
_/home/travis/gopath/src/github.com/coreos/etcd/integration
_/home/travis/gopath/src/github.com/coreos/etcd/lease
_/home/travis/gopath/src/github.com/coreos/etcd/lease/leasehttp
_/home/travis/gopath/src/github.com/coreos/etcd/lease/leasepb
_/home/travis/gopath/src/github.com/coreos/etcd/mvcc
_/home/travis/gopath/src/github.com/coreos/etcd/mvcc/backend
_/home/travis/gopath/src/github.com/coreos/etcd/mvcc/mvccpb
_/home/travis/gopath/src/github.com/coreos/etcd/pkg/adt
_/home/travis/gopath/src/github.com/coreos/etcd/pkg/contention
_/home/travis/gopath/src/github.com/coreos/etcd/pkg/cors
_/home/travis/gopath/src/github.com/coreos/etcd/pkg/crc
_/home/travis/gopath/src/github.com/coreos/etcd/pkg/expect
_/home/travis/gopath/src/github.com/coreos/etcd/pkg/fileutil
_/home/travis/gopath/src/github.com/coreos/etcd/pkg/flags
_/home/travis/gopath/src/github.com/coreos/etcd/pkg/httputil
_/home/travis/gopath/src/github.com/coreos/etcd/pkg/idutil
_/home/travis/gopath/src/github.com/coreos/etcd/pkg/ioutil
_/home/travis/gopath/src/github.com/coreos/etcd/pkg/logutil
_/home/travis/gopath/src/github.com/coreos/etcd/pkg/mock/mockstorage
_/home/travis/gopath/src/github.com/coreos/etcd/pkg/mock/mockstore
_/home/travis/gopath/src/github.com/coreos/etcd/pkg/mock/mockwait
_/home/travis/gopath/src/github.com/coreos/etcd/pkg/monotime
_/home/travis/gopath/src/github.com/coreos/etcd/pkg/netutil
_/home/travis/gopath/src/github.com/coreos/etcd/pkg/osutil
_/home/travis/gopath/src/github.com/coreos/etcd/pkg/pathutil
_/home/travis/gopath/src/github.com/coreos/etcd/pkg/pbutil
_/home/travis/gopath/src/github.com/coreos/etcd/pkg/report
_/home/travis/gopath/src/github.com/coreos/etcd/pkg/runtime
_/home/travis/gopath/src/github.com/coreos/etcd/pkg/schedule
_/home/travis/gopath/src/github.com/coreos/etcd/pkg/stringutil
_/home/travis/gopath/src/github.com/coreos/etcd/pkg/testutil
_/home/travis/gopath/src/github.com/coreos/etcd/pkg/tlsutil
_/home/travis/gopath/src/github.com/coreos/etcd/pkg/transport
_/home/travis/gopath/src/github.com/coreos/etcd/pkg/types
_/home/travis/gopath/src/github.com/coreos/etcd/pkg/wait
_/home/travis/gopath/src/github.com/coreos/etcd/proxy/grpcproxy
_/home/travis/gopath/src/github.com/coreos/etcd/proxy/grpcproxy/cache
_/home/travis/gopath/src/github.com/coreos/etcd/proxy/httpproxy
_/home/travis/gopath/src/github.com/coreos/etcd/proxy/tcpproxy
_/home/travis/gopath/src/github.com/coreos/etcd/raft
_/home/travis/gopath/src/github.com/coreos/etcd/raft/raftpb
_/home/travis/gopath/src/github.com/coreos/etcd/raft/rafttest
_/home/travis/gopath/src/github.com/coreos/etcd/rafthttp
_/home/travis/gopath/src/github.com/coreos/etcd/snap
_/home/travis/gopath/src/github.com/coreos/etcd/snap/snappb
_/home/travis/gopath/src/github.com/coreos/etcd/store
_/home/travis/gopath/src/github.com/coreos/etcd/tools/benchmark
_/home/travis/gopath/src/github.com/coreos/etcd/tools/benchmark/cmd
_/home/travis/gopath/src/github.com/coreos/etcd/tools/etcd-dump-db
_/home/travis/gopath/src/github.com/coreos/etcd/tools/etcd-dump-logs
_/home/travis/gopath/src/github.com/coreos/etcd/tools/functional-tester/etcd-agent
_/home/travis/gopath/src/github.com/coreos/etcd/tools/functional-tester/etcd-agent/client
_/home/travis/gopath/src/github.com/coreos/etcd/tools/functional-tester/etcd-runner
_/home/travis/gopath/src/github.com/coreos/etcd/tools/functional-tester/etcd-tester
_/home/travis/gopath/src/github.com/coreos/etcd/tools/local-tester/bridge
_/home/travis/gopath/src/github.com/coreos/etcd/version
_/home/travis/gopath/src/github.com/coreos/etcd/wal
_/home/travis/gopath/src/github.com/coreos/etcd/wal/walpb
```

It lists a whole bunch of deps since Travis copies this before we run the tests.

So I grep by `github.com/coreos/etcd` to exclude the external depencies.
Then now we have

```
_/home/travis/gopath/src/github.com/coreos/etcd/alarm
_/home/travis/gopath/src/github.com/coreos/etcd/auth
_/home/travis/gopath/src/github.com/coreos/etcd/auth/authpb
_/home/travis/gopath/src/github.com/coreos/etcd/client
...
```

Next commands will just replace `s/_/home/travis/gopath/src/github.com/coreos/etcd///` just to get etcd packages.

**Comment by gyuho** [comment]
should be fixed https://travis-ci.org/coreos/etcd/jobs/182976362#L503.

removed `go list` command in test script.

**Comment by heyitsanthony** [comment]
repush to kick off jenkins?

**Comment by gyuho** [comment]
@heyitsanthony just repushed and seems like it passes now?

**Comment by heyitsanthony** [comment]
lgtm. Thanks!

**Pull request merged**